File: web/src/models/pet.ts
export type PetMood = 'happy' | 'calm' | 'hungry' | 'tired' | 'upset' | 'sick';

export interface PetUserProfile {
  id: string;
  name: string;
  petIds: string[];
  achievements: Achievement[];
  createdAt: string;
}

export interface PetInstance {
  id: string;
  userId: string;
  animalTypeId: string;
  name: string;
  birthday: string;
  lastUpdatedAt: string;
  speedMultiplier: number;
  mood: PetMood;
  hunger: number;
  thirst: number;
  energy: number;
  happiness: number;
  health: number;
  hygiene: number;
  stress: number;
  isSick: boolean;
  outfitIds: string[];
  friendIds: string[];
}

export interface Achievement {
  id: string;
  title: string;
  description: string;
  unlockedAt?: string;
}

export interface OutfitItem {
  id: string;
  name: string;
  slot: string;
  icon: string;
  imageUrl?: string;
  allowedCategoryIds?: string[];
  note: string;
}

export interface FriendPet {
  id: string;
  name: string;
  animalTypeId: string;
  ownerName: string;
  mood: string;
  visitCount: number;
}

File: Sources/PetPet/Models/Pet.swift
import Foundation

struct Pet: Codable, Identifiable {
    let id: UUID
    var typeID: String
    var name: String
    var birthday: Date
    var hunger: Double       // 0.0–1.0
    var thirst: Double
    var happiness: Double
    var health: Double
    var isSick: Bool
    var speedMultiplier: Double  // 1 = 实时，10 = 10倍速

    var growthStage: GrowthStage {
        GrowthStage.current(birthday: birthday, speedMultiplier: speedMultiplier)
    }

    init(typeID: String, name: String, speedMultiplier: Double = 10) {
        self.id = UUID()
        self.typeID = typeID
        self.name = name
        self.birthday = Date()
        self.hunger = 0.8
        self.thirst = 0.8
        self.happiness = 0.9
        self.health = 1.0
        self.isSick = false
        self.speedMultiplier = speedMultiplier
    }
}
